src/client/Microsoft.Identity.Client/Platforms/iOS/SystemWebview/ASWebAuthenticationPresentationContextProviderWindow.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using AuthenticationServices;
+using Foundation;
+using UIKit;
+
+namespace Microsoft.Identity.Client.Platforms.iOS.SystemWebview
+{
+    internal class ASWebAuthenticationPresentationContextProviderWindow : NSObject, IASWebAuthenticationPresentationContextProviding
+    {
+        public UIWindow GetPresentationAnchor(ASWebAuthenticationSession session)
+        {
+            return UIApplication.SharedApplication.KeyWindow;
```

Comment:
>KeyWindow [](start = 51, length = 9)

Will this work when users pass in their own window? 
What impact will this change have for embedded web view?

Reply:
this won't impact embedded webview. Not sure if they pass in their own window, i'd have to test that out tomorrow.

Reply:
ASWebAuthSession is just for system browser

Reply:
thanks. 

---
In reply to: [331279014](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1414#discussion_r331279014) [](ancestors = 331279014)

Reply:
I would like to understand if there will be any potential issues passing in a window... for now assuming that is similar to embedded webview behavior.

---
In reply to: [331278925](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1414#discussion_r331278925) [](ancestors = 331278925)

Reply:
sounds good. will have an answer for you tomorrow.